### PR TITLE
Mosip 12368

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java [hunk 1]
```diff
@@ -1,341 +1,465 @@
...
+        String finalObjectName=null;
+   	    
+   	   if(useAccountAsBucketname) {
+   		  finalObjectName = ObjectStoreUtil.getName(container,source, process, objectName);
+   		
+   	   }else {
+   		 finalObjectName = ObjectStoreUtil.getName(source, process, objectName);
+   		}
+       
+        return addObjectMetaData(account, container, source, process, finalObjectName, meta);
```
Comment (vishwa-vyom) [inline kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java:207]: This method is already concatinating the finalObjectName, but why are we doing it again in this method?

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java [hunk 2]
```diff
@@ -1,341 +1,465 @@
...
+			}
+		  }
+		}
+		}catch(Exception e){
+			LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while getTags for : " + container,
+					ExceptionUtils.getStackTrace(e));
+			throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(),
+					OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+		}
+		return objectTags;
```
Comment (vishwa-vyom) [inline kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java]: If we throwning the above exception, then can we just return inside the try block itself ?

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java [hunk 3]
```diff
@@ -1,341 +1,465 @@
...
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.BucketTaggingConfiguration;
```
Comment (vishwa-vyom) [inline kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java]: Can we remove the unused imports ?

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/util/ObjectStoreUtil.java [hunk 4]
```diff
@@ -1,20 +1,33 @@
...
+        if (StringUtils.isNotEmpty(source))
+            finalObjectName = source + SEPARATOR;
+        if (StringUtils.isNotEmpty(process))
+            finalObjectName = finalObjectName + process + SEPARATOR;
+
+        finalObjectName = finalObjectName + objectName;
+
+        return finalObjectName;
+    }
+    public static String getName(String Container,String source, String process, String objectName) {
```
Comment (vishwa-vyom) [inline kernel/object-store/src/main/java/io/mosip/commons/khazana/util/ObjectStoreUtil.java]: Container should be small c

**Review by vishwa-vyom — COMMENTED** [review]
only minor items, lets' close and then we should be good to merge

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java (hunk 1 above)
Comment (sowmya695) [reply kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java:207]: In previous release also it is concatinating so i have kept as it is ,it should not affect existing flow

File: kernel/object-store/src/main/java/io/mosip/commons/khazana/util/ObjectStoreUtil.java (hunk 4 above)
Comment (sowmya695) [reply kernel/object-store/src/main/java/io/mosip/commons/khazana/util/ObjectStoreUtil.java]: Fixed

**Pull request merged**